packages/origin-247-certificate/src/certificateForUnitTests.service.ts

```diff
@@ -0,0 +1,120 @@
+import { CertificateService } from './certificate.service';
+import { EventBus } from '@nestjs/cqrs';
+import {
+    ICertificate,
+    IClaimCommand,
+    ITransferCommand,
+    ISuccessResponse,
+    IIssuedCertificate,
+    IIssueCommandParams
+} from './types';
+import { Injectable } from '@nestjs/common';
+import { CertificatePersistedEvent } from './externals';
+
+type PublicPart<T> = { [K in keyof T]: T[K] };
+
+@Injectable()
+export class CertificateForUnitTestsService<T> implements PublicPart<CertificateService<T>> {
+    private serial = 0;
+    private db: ICertificate<T>[] = [];
+
+    constructor(private readonly eventBus: EventBus) {}
+
+    public async getAll(): Promise<ICertificate<T>[]> {
+        return [...this.db];
+    }
+
+    public async getById(id: number): Promise<ICertificate<T> | null> {
+        return this.db.find((c) => c.id === id) ?? null;
+    }
+
+    public async issue(params: IIssueCommandParams<T>): Promise<IIssuedCertificate<T>> {
+        this.serial += 1;
+
+        const certificate: ICertificate<T> = {
+            claims: [],
+            claimers: {},
+            creationBlockHash: '',
+            creationTime: Math.floor(Date.now() / 1000),
+            deviceId: params.deviceId,
+            generationStartTime: Math.floor(params.toTime.getTime() / 1000),
+            generationEndTime: Math.floor(params.toTime.getTime() / 1000),
+            id: this.serial,
+            tokenId: this.serial,
```

Comment:
TokenId no longer exists if you are using the latest canary release from yesterday?

Reply:
No I'm not. I will update it

Reply:
removed